lib/internal/bootstrap/node.js

```diff
@@ -531,6 +531,14 @@
       // call that threw and was never cleared. So clear it now.
       clearDefaultTriggerAsyncId();
 
+      try {
+        // if node-report is enabled, call into its handler to see
+        // whether it is interested in handling the situation.
+        const report = internalBinding('node_report');
+        if (report !== null && report.onUnCaughtException !== null)
+          report.onUnCaughtException(er);
+      } catch (e) {}  // nothing to do; node-report is unavailable.
```

Comment:
style nit: You can omit the `(e)`

The comment seems to hint at the possibility that `onUnCaughtException` can’t throw by itself … is that correct? If so, can we move it to outside of the try/catch?

And again, I can’t reply to your earlier comments, but…

> Looks like the original exception context is lost when we return from JS to C++ and run through my code (though I am passing the `err` object I am obtaining. don't know where I am messing things, will you please have a look and advise?

I’m not sure – can you explain what you mean by “exception context”? The stack trace should still be intact, I think…?

Reply:
Yes, I mean the callstack is not in the failing context:

With the change in `lib/internal/bootstrap/node.js` , my report has this call stack:

```gdb
process._fatalException (internal/bootstrap/node.js:539:18)
```

as opposed to an expected:

```gdb
Object.<anonymous> (/home/user/foo.js:1:1)
Module._compile (internal/modules/cjs/loader.js:1:1)
Object.Module._extensions..js (internal/modules/cjs/loader.js:1:1)
Module.load (internal/modules/cjs/loader.js:1:1)
tryModuleLoad (internal/modules/cjs/loader.js:1:1)
Function.Module._load (internal/modules/cjs/loader.js:1:1)
Function.Module.runMain (internal/modules/cjs/loader.js:1:1)
startup (internal/bootstrap/node.js:1:1)
bootstrapNodeJSCore (internal/bootstrap/node.js:1:1)
```

so looks like the `error` object received by `process._fatalException` is lost somewhere on the way!

Reply:
Okay, I see – maybe we should pass through the `Message` object from the fatal error handler (or from the exception) for the error to `PrintStackFromStackTrace`?

Reply:
> The comment seems to hint at the possibility that onUnCaughtException can’t throw by itself … is that correct? If so, can we move it to outside of the try/catch?

try-catch block: if node is built without node-report (configure --without-node-report option)  then the `internalBinding('node-report')` will fail, so I am using the try-catch block to cover such cases, and is absolutely necessary (unless we have other JS semantics to figure out the presence of a module.) So with that approach, I guess we cannot take it out of the try-catch block.

Reply:
> we should pass through the Message object from the fatal error handler (or from the exception) for the error to PrintStackFromStackTrace?

Sorry, wasn't able to convert this into a working code. Can you please elaborate on this?

Reply:
this is addressed now.